animations/src/main/java/com/chiralbehaviors/CoRE/meta/models/ModelImpl.java

```diff
@@ -144,6 +147,23 @@ public ModelImpl(EntityManagerFactory emf) {
                                                           updatedBy);
     }
 
+    /* (non-Javadoc)
+     * @see com.chiralbehaviors.CoRE.meta.Model#executeAs(com.chiralbehaviors.CoRE.security.AuthenticatedPrincipal, java.util.concurrent.Callable)
+     */
+    @Override
+    public <V> V executeAs(AuthenticatedPrincipal principal,
+                           Callable<V> function) throws Exception {
+        V value = null;
+        AuthenticatedPrincipal previous = currentPrincipal.get();
```

Comment:
Our entire trigger logic is tied to our wrapper of the entity manager.    And we have only one model instance per entity manager.  You can't create a model without an EMF, and that wraps the instance of the EM used.  So there's no way out Model <-> EM are tightly bound.  

As to the stack, it can be invoked recursively and the state would be on the literal java stack at that point.  So, as deep as your stack can get, if you make repeated recursive calls to this function.